lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -265,6 +265,9 @@ && isIntegerOrDecimalPrimitive(primitiveType)) {
             return createColumnReader(field, valueDecoders::getBinaryDecoder, BINARY_ADAPTER, memoryContext);
         }
         if ((VARBINARY.equals(type) || VARCHAR.equals(type)) && primitiveType == FIXED_LEN_BYTE_ARRAY) {
+            if (annotation instanceof DecimalLogicalTypeAnnotation) {
```

Comment:
@raunaqmorarka IIUC this is to restrict reading decimal columns as varchar right ?

Reply:
yes, this was being allow unintentionally by the code here, it should be blocked since the conversion hasn't been implemented in the parquet reader code